src/context/service-worker-context.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useEffect, useState } from 'react'
+import { registerServiceWorker } from '../service-worker-utils'
+
+export const ServiceWorkerContext = createContext({
+  isServiceWorkerRegistered: false
+})
+
+export const ServiceWorkerProvider = ({ children }): JSX.Element => {
+  const [isServiceWorkerRegistered, setIsServiceWorkerRegistered] = useState(false)
+
+  useEffect(() => {
+    if (isServiceWorkerRegistered) {
+      return
+    }
+    async function doWork (): Promise<void> {
+      const registration = await navigator.serviceWorker.getRegistration()
+
+      if (registration != null) {
```

Comment:
I've seen you use the loose comparison with null a lot. I presume it's to catch `undefined` too?

Generally speaking though, why not flip the cases here and check for a truthy registration object? It's a bit easier to reason about the code that way

Reply:
>  I presume it's to catch undefined too?

Yea, that's correct.

> why not flip the cases here and check for a truthy registration object?

I am checking for a truthy registration object. if getRegistration is not null, it's a registration object. if you're talking about doing something like instanceof.. it's just a little more verbose.

Also, I try to adhere to guard clause patterns where the earliest "exit" is done first. This could probably be cleaned up a little so we're not doing two `.update()` calls.